app/api/models/domain/ChargeReference.scala

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api.models.domain
+
+case class ChargeReference(chargeReference: String)
```

Comment:
Very minor thing but when you have a wrapper class for a value and you name the value after the wrapper class, then when you want to access the field's value inside an `ChargeReference` object that will typically be called "chargeReference" you can end up with code like `chargeReference.chargeReference` which might raise eyebrows. Is this a reference to another `ChargeReference` instance?

You could instead do:
```suggestion
case class ChargeReference(value: String)
```
(but I notice that in the connector you call the `ChargeReference` value so then you'd have `value.value` so that might need to be renamed also 😆).